libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRouteProcessor.java

```diff
@@ -134,10 +139,40 @@ private void updateIntersections() {
     currentIntersectionDistances = createDistancesToIntersections(currentStepPoints, currentIntersections);
   }
 
-  private void addUpcomingStepPoints(RouteProgress.Builder progressBuilder) {
-    if (upcomingStepPoints != null && !upcomingStepPoints.isEmpty()) {
-      progressBuilder.upcomingStepPoints(upcomingStepPoints);
+  private void addTotalTraveled(MapboxNavigator navigator, RouteProgress.Builder progressBuilder) {
+    List<TrackedRoute> history = navigator.retrieveRouteHistory();
+    double totalDistanceTraveled = 0d;
+    float totalDurationTraveled = 0f;
+    float totalFractionTraveled = 0f;
+    double totalDistance = 0f;
```

Comment:
`0d`? Though should be optional, just make it consistent with `totalDistanceTraveled`